cli/src/cli_util.rs

```diff
@@ -1090,6 +1097,28 @@ impl WorkspaceCommandHelper {
         }
     }
 
+    /// Acquires a lock for git import/export operations if the workspace is
+    /// colocated with Git. Returns a token that can be passed to functions
+    /// that need to import from or export to Git. For non-colocated repos,
+    /// returns a token with no lock inside.
+    fn lock_git_import_export(&self) -> Result<GitImportExportLock, CommandError> {
+        let lock = if self.working_copy_shared_with_git {
+            let lock_path = self.workspace.repo_path().join("git_import_export.lock");
+            Some(
+                jj_lib::lock::FileLock::lock(lock_path.clone()).map_err(|err| {
+                    user_error(format!(
+                        "Failed to take lock for Git import/export: {}: {}",
+                        lock_path.display(),
+                        err.err
+                    ))
```

Comment:
nit: `user_error_with_message("Failed to take lock for Git import/export", err)`, which should print why the lock couldn't be obtained.